flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobRecoveryITCase.java

```diff
@@ -21,62 +21,60 @@
 import org.apache.flink.util.TestLogger;
 
-import org.junit.ClassRule;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.concurrent.CompletableFuture;
-
-import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertThat;
+import java.util.concurrent.Executor;
 
 /** Tests for the recovery of task failures. */
 public class JobRecoveryITCase extends TestLogger {
 
-    private static final int NUM_TMS = 1;
-    private static final int SLOTS_PER_TM = 11;
-    private static final int PARALLELISM = NUM_TMS * SLOTS_PER_TM;
+    private static final int PARALLELISM = 1;
 
-    @ClassRule
-    public static final MiniClusterResource MINI_CLUSTER_RESOURCE =
-            new MiniClusterResource(
-                    new MiniClusterResourceConfiguration.Builder()
-                            .setNumberTaskManagers(NUM_TMS)
-                            .setNumberSlotsPerTaskManager(SLOTS_PER_TM)
-                            .build());
+    private static class MyHaServices extends EmbeddedHaServicesWithLeadershipControl {
 
-    @Test
-    public void testTaskFailureRecovery() throws Exception {
-        runTaskFailureRecoveryTest(createjobGraph(false));
+        public MyHaServices(Executor executor) {
+            super(executor);
+        }
     }
 
     @Test
-    public void testTaskFailureWithSlotSharingRecovery() throws Exception {
-        runTaskFailureRecoveryTest(createjobGraph(true));
-    }
-
-    private void runTaskFailureRecoveryTest(final JobGraph jobGraph) throws Exception {
-        final MiniCluster miniCluster = MINI_CLUSTER_RESOURCE.getMiniCluster();
-
-        miniCluster.submitJob(jobGraph).get();
-
-        final CompletableFuture<JobResult> jobResultFuture =
-                miniCluster.requestJobResult(jobGraph.getJobID());
-
-        assertThat(jobResultFuture.get().isSuccess(), is(true));
+    public void testTaskFailureRecovery() throws Exception {
+        final TestingMiniClusterConfiguration configuration =
+                new TestingMiniClusterConfiguration.Builder()
+                        //                .setNumberDispatcherResourceManagerComponents(2)
+                        .setNumTaskManagers(1)
+                        .setNumTaskManagers(1)
+                        .build();
+        final HighAvailabilityServices haServices =
+                new MyHaServices(TestingUtils.defaultExecutor());
+
+        try (TestingMiniCluster miniCluster =
+                new TestingMiniCluster(configuration, () -> haServices)) {
+            miniCluster.start();
+            JobGraph jobGraph = createjobGraph(true);
+            miniCluster.submitJob(jobGraph).get();
+            final CompletableFuture<JobResult> jobResultFuture =
+                    miniCluster.requestJobResult(jobGraph.getJobID());
+            JobResult jobResult = jobResultFuture.get();
+            System.out.println(jobResult.getApplicationStatus());
+        }
```

Comment:
It looks as if we should revert this change.

Reply:
👍